chunky/src/java/se/llbit/chunky/renderer/scene/PathTracer.java

```diff
@@ -95,12 +95,12 @@ public static boolean pathTrace(Scene scene, Ray ray, WorkerState state, int add
       Material currentMat = ray.getCurrentMaterial();
       Material prevMat = ray.getPrevMaterial();
 
-      if (!scene.stillWater && ray.n.y != 0 &&
+      if (!scene.stillWater && ray.getN().y != 0 &&
           ((currentMat.isWater() && prevMat == Air.INSTANCE)
               || (currentMat == Air.INSTANCE && prevMat.isWater()))) {
         WaterModel.doWaterDisplacement(ray);
         if (currentMat == Air.INSTANCE) {
-          ray.n.y = -ray.n.y;
+          ray.invertNormal();
```

Comment:
Previously, only `y` was inverted. Now all components of the normal are inverted. Is this intentional, was the code wrong before?

Reply:
It was intentional because it makes more sense (I think) like this. If you think as normal mapping simulating displaced geometry, you want to invert the whole normal when working with the underside of the surface.
But in practice, it didn't have any perverse effect, the worse being that technically if you looked at both side of some water surface, they would not correspond to each other